plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -306,31 +310,50 @@ protected void renameTable(ConnectorSession session, String catalogName, String
-                "sp_rename %s, %s",
-                singleQuote(catalogName, remoteSchemaName, remoteTableName),
-                singleQuote(newRemoteTableName));
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String newSchemaName = newTable.getSchemaName();
+            String newTableName = newTable.getTableName();
+            verifyTableName(connection.getMetaData(), newTableName);
+            ConnectorIdentity identity = session.getIdentity();
+            String newRemoteSchemaName = identifierMapping.toRemoteSchemaName(identity, connection, newSchemaName);
+            String newRemoteTableName = identifierMapping.toRemoteTableName(identity, connection, newRemoteSchemaName, newTableName);
+
+            String fullTableFromName = DOT_JOINER.join(catalogName, schemaName, quoteEscapeForSqlServer(tableName));
+
+            try (CallableStatement renameTable = connection.prepareCall("exec sp_rename ?, ?")) {
+                renameTable.setString(1, fullTableFromName);
+                renameTable.setString(2, newRemoteTableName);
+                renameTable.execute();
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
     }
 
     @Override
-    protected String renameColumnSql(JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newRemoteColumnName)
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
     {
         RemoteTableName remoteTableName = handle.asPlainTable().getRemoteTableName();
-        return format(
-                "sp_rename %s, %s, 'COLUMN'",
-                singleQuote(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), "[" + escape(jdbcColumn.getColumnName()) + "]"),
-                "[" + newRemoteColumnName + "]");
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String columnFrom = DOT_JOINER.join(remoteTableName.getCatalogName().orElseThrow(), remoteTableName.getSchemaName().orElseThrow(), quoteEscapeForSqlServer(remoteTableName.getTableName()), quoteEscapeForSqlServer(jdbcColumn.getColumnName()));
+            String newRemoteColumnName = identifierMapping.toRemoteColumnName(connection, newColumnName);
+            verifyColumnName(connection.getMetaData(), newRemoteColumnName);
+
+            try (CallableStatement renameColumn = connection.prepareCall("exec sp_rename ?, ?, 'COLUMN'")) {
+                renameColumn.setString(1, columnFrom);
+                renameColumn.setString(2, newRemoteColumnName);
+                renameColumn.execute();
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
     }
 
-    private static String escape(String name)
+    private String quoteEscapeForSqlServer(String name)
```

Comment:
nit: `quoteForSqlServer` might be clearer name. Proper quoting includes escaping the quotes.